test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	siteEditorStyleVariations: async (
+		{ admin, editor, page, pageUtils, requestUtils },
+		use
+	) => {
+		await use(
+			new SiteEditorStyleVariations( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+				requestUtils,
+			} )
+		);
+	},
+} );
+
+test.describe( 'Global styles variations', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme(
+			'gutenberg-test-themes/style-variations'
+		);
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'should have three variations available with the first one being active', async ( {
+		admin,
+		page,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await page.click( 'role=button[name="Styles"i]' );
+		await page.click( 'role=button[name="Browse styles"i]' );
+
+		// TODO: instead of locating these elements by class,
+		//  we could update the source code to group them in a <section> or other container,
+		//  then add `aria-labelledby` and `aria-describedby` etc to provide accessible information,
+		const variations = page.locator(
+			'.edit-site-global-styles-variations_item'
+		);
+
+		await expect( variations ).toHaveCount( 3 );
+
+		await expect( variations.first() ).toHaveAttribute(
+			'aria-current',
+			'true'
+		);
+		await expect( variations.nth( 1 ) ).toHaveAttribute(
+			'aria-current',
+			'false'
+		);
+		await expect( variations.nth( 2 ) ).toHaveAttribute(
+			'aria-current',
+			'false'
+		);
+	} );
+
+	test( 'should apply preset colors and font sizes in a variation', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+		await page.click( 'role=button[name="Colors styles"i]' );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] [data-testid="background-color-indicator"]'
+			)
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors text styles"] [data-testid="text-color-indicator"]'
```

Comment:
There's actually a [selector](https://playwright.dev/docs/selectors#id-data-testid-data-test-id-data-test-selectors) specifically for data-testid 😆 :

```suggestion
				'css=[aria-label="Colors text styles"] data-testid=text-color-indicator'
```

Also, any reason why we're using a css selector for `'css=[aria-label="Colors text styles"]` rather than a role selector?

Ditto to one above.

Reply:
Oh nice!

`'role=button[name="Colors background styles"i] >> data-testid=background-color-indicator'` seems to work.